From 62f25c7293254bcb465c6c85bda61bff4630b8c8 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 1 May 2021 00:57:30 -0700
Subject: [PATCH] Use __typeof__ instead of typeof

The compiler may not define typeof in strict C mode to prevent
namespace polution.
---
 tools/imagetool.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/imagetool.h b/tools/imagetool.h
index 2801ea9e9f..bbdf82877b 100644
--- a/tools/imagetool.h
+++ b/tools/imagetool.h
@@ -26,7 +26,7 @@
 #define ARRAY_SIZE(x)		(sizeof(x) / sizeof((x)[0]))
 
 #define __ALIGN_MASK(x, mask)	(((x) + (mask)) & ~(mask))
-#define ALIGN(x, a)		__ALIGN_MASK((x), (typeof(x))(a) - 1)
+#define ALIGN(x, a)		__ALIGN_MASK((x), (__typeof__(x))(a) - 1)
 
 #define IH_ARCH_DEFAULT		IH_ARCH_INVALID
 
-- 
2.31.1

